packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -82,12 +98,16 @@ def _zipfile_single_file_extract_worker(
         if is_dir:
             destination_path.mkdir(parents=True, exist_ok=True)
             return destination_path
-
-        with zf.open(name=file_in_archive) as zip_fp:
-            with open(destination_path, "wb") as destination_fp:
-                for chunk in _read_in_chunks(zip_fp):
-                    destination_fp.write(chunk)
-                return destination_path
+        desc = f"decompressing {zip_file_path}:{file_in_archive.filename} -> {destination_path}\n"
+        with zf.open(name=file_in_archive) as zip_fp, destination_path.open(
+            "wb"
+        ) as dest_fp, tqdm(
+            total=file_in_archive.file_size, desc=desc, **_TQDM_FILE_OPTIONS
+        ) as pbar:
```

Comment:
So even if this is one out of 100 files it still reports the correct progress? This feels like you report the progress file per file?

Reply:
this is the progress per file